django/db/migrations/autodetector.py

```diff
@@ -810,22 +795,7 @@ def _generate_added_field(self, app_label, model_name, field_name):
         # Fields that are foreignkeys/m2ms depend on stuff
         dependencies = []
         if field.remote_field and field.remote_field.model:
-            # Account for FKs to swappable models
-            swappable_setting = getattr(field, 'swappable_setting', None)
-            if swappable_setting is not None:
-                dep_app_label = "__setting__"
-                dep_object_name = swappable_setting
-            else:
-                dep_app_label = field.remote_field.model._meta.app_label
-                dep_object_name = field.remote_field.model._meta.object_name
-            dependencies = [(dep_app_label, dep_object_name, None, True)]
-            if getattr(field.remote_field, "through", None) and not field.remote_field.through._meta.auto_created:
-                dependencies.append((
-                    field.remote_field.through._meta.app_label,
-                    field.remote_field.through._meta.object_name,
-                    None,
-                    True,
-                ))
+            self._add_dependecies_for_foreign_key(field, dependencies)
```

Comment:
Could you use `dependencies.extend(self._add_dependecies_for_foreign_key(field))` and completely remove the `dependencies` argument instead?